qemu/tests/boot_with_machine_types.py

```diff
@@ -0,0 +1,47 @@
+import logging
+
+from virttest import error_context
+from virttest import utils_misc
+from virttest import env_process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    boot with different machine type:
+    1) get supported machine type
+    2) boot guest with different machine types
+
+    :param test: kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    error_context.context("Get supported machine type", logging.info)
+    qemu_binary = utils_misc.get_qemu_binary(params)
+    supported_machine_types, supported_system_name = utils_misc.get_support_machine_type(qemu_binary)
+    machine_types = []
+    machine_type_mapping = {"pc": ["i440FX", "RHEL 6"], "q35": ["Q35"], "pseries": ["pSeries"], "arm64-pci:virt": ["ARM"], "arm64-mmio:virt": ["ARM"], "s390-ccw-virtio": ["S390"]}
+    for m_type, s_name in zip(supported_machine_types, supported_system_name):
+        for item in machine_type_mapping[params["machine_type"]]:
+            if item in s_name:
+                machine_types.append(m_type)
+    logging.info("Actual supproted machine types are: " + str(machine_types))
```

Comment:
%s/supproted/supported
Also let us format the list as ','.join(map(str, machine_types)) for better logging.